File: backend/models/expense.py
import uuid
from datetime import date, datetime

from sqlalchemy import Boolean, Date, DateTime, ForeignKey, Numeric, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func

from database import Base


class CaseExpense(Base):
    __tablename__ = "case_expenses"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid())
    case_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("cases.id", ondelete="CASCADE"), nullable=False, index=True
    )
    description: Mapped[str] = mapped_column(String, nullable=False)
    amount: Mapped[float] = mapped_column(Numeric(10, 2), nullable=False)
    date: Mapped[date] = mapped_column(Date, nullable=False)
    category: Mapped[str] = mapped_column(String, nullable=False)
    receipt_doc_id: Mapped[uuid.UUID | None] = mapped_column(UUID(as_uuid=True), ForeignKey("documents.id"))
    recoverable: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

File: frontend/src/types/expense.ts
// Mirrors backend/schemas/expenses.py

export interface ExpenseOut {
  id: string
  description: string
  amount: number
  date: string
  category: string
  receipt_doc_id: string | null
  recoverable: boolean
  created_at: string
}

export interface ExpenseCreate {
  description: string
  amount: number
  date: string
  category: string
  receipt_doc_id?: string | null
  recoverable?: boolean
}

export type ExpenseUpdate = Partial<ExpenseCreate>
